[clang-tidy] Improved readability-bool-conversion to be more consistent when using parentheses

felix642 opened the pull request:
Fixes #71852

felix642 committed:
[clang-tidy] Improved readability-bool-conversion to be more consistent when using parentheses

Fixes #71852

firewave commented:
Another PR for this was opened a few hours ago: #71848.

felix642 commented:
Hi @firewave, I think you are referencing a different issue. If I test #71852 with PR #72050 I do not get the expected behavior.

firewave commented:
> Hi @firewave, I think you are referencing a different issue. If I test #71852 with PR #72050 I do not get the expected behavior.

Of course you are right. I missed there being two different issues. Sorry about the noise.

PiotrZSL commented on clang-tools-extra/docs/ReleaseNotes.rst:
```diff
@@ -412,6 +412,10 @@ Changes in existing checks
   do-while loops into account for the `AllowIntegerConditions` and
   `AllowPointerConditions` options.
 
+- Improved :doc:`readability-implicit-bool-conversion
```
merge release notes, one entry for a check, one sentence for a change

PiotrZSL reviewed (approved):
Description for a change (commit / PR) needed.
Except that looks fine.

felix642 committed:
fixup! [clang-tidy] Improved readability-bool-conversion to be more consistent when using parentheses

Updated ReleaseNotes.rst

felix642 committed:
amend! [clang-tidy] Improved readability-bool-conversion to be more consistent when using parentheses

[clang-tidy] Improved readability-bool-conversion to be more consistent

The check would sometimes suggest replacing the value with a literal,
but if parentheses were added it would instead suggest to static_cast
the value. With this commit the check always suggest to replace with a
literal when possible.'

Fixes #71852
